daemon/logger/jsonfilelog/jsonfilelog_test.go

```diff
@@ -54,6 +54,51 @@ func TestJSONFileLogger(t *testing.T) {
 	}
 }
 
+func TestJSONFileLoggerWithTags(t *testing.T) {
+	cid := "a7317399f3f857173c6179d44823594f8294678dea9999662e5c625b5a1c7657"
+	cname := "test-container"
+	tmp, err := ioutil.TempDir("", "docker-logger-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+	filename := filepath.Join(tmp, "container.log")
+	l, err := New(logger.Info{
+		Config: map[string]string{
+			"tag": "{{.ID}}/{{.Name}}", // first 10 characters of ContainerID and full ContainerName
```

Comment:
This is my first time trying a catchup merge here. I hope I didn't screw it up. Brought up to the latest master and changed the comment to 'first 12 characters ...'. I'll ping you directly in the AM as long as the build is successful.